go/adbc/driver/flightsql/flightsql_adbc_server_test.go

```diff
@@ -1173,9 +1181,11 @@ type CookieTests struct {
 }
 
 func (suite *CookieTests) SetupSuite() {
-	suite.DoSetupSuite(&CookieTestServer{}, nil, map[string]string{
+	ts := &CookieTestServer{}
+	suite.DoSetupSuite(ts, nil, map[string]string{
 		driver.OptionCookieMiddleware: adbc.OptionValueEnabled,
 	})
+	ts.addr = "grpc://" + suite.s.Addr().String()
 }
 
 func (suite *CookieTests) TestCookieUsage() {
```

Comment:
Can we add a specific test to make sure the token/cookies are copied?

Reply:
The modification i made to the test https://github.com/apache/arrow-adbc/pull/1497/files#diff-fff4d523df1762b32c2cd86dbe03bd2ec80fe856fc91471f24604909486824efL1098 does this

Reply:
Hmm, the client cache isn't shared between metadata/data? Looks like not. That may be something we want to fix in the future so this test may be a little unstable in that case.